plugins/lsp/language-servers/efmls-configs.nix

```diff
@@ -236,43 +99,42 @@ in
             if cfg.externallyManagedPackages == "all" then
               _: false
             else
-              toolName: !(elem toolName cfg.externallyManagedPackages);
-          partition = lists.partition partitionFn tools;
+              toolName: !(lib.elem toolName cfg.externallyManagedPackages);
+          partition = lib.lists.partition partitionFn tools;
         in
         {
           nixvim = partition.right;
           external = partition.wrong;
         };
 
-      nixvimPkgs = lists.partition (v: hasAttr v cfg.toolPackages) pkgsForTools.nixvim;
+      nixvimPkgs = lib.lists.partition (v: lib.hasAttr v cfg.toolPackages) pkgsForTools.nixvim;
 
-      mkToolOption =
+      mkToolValue =
         kind: opt:
         map (
-          tool: if isString tool then helpers.mkRaw "require 'efmls-configs.${kind}.${tool}'" else tool
-        ) (toList opt);
+          tool: if lib.isString tool then helpers.mkRaw "require 'efmls-configs.${kind}.${tool}'" else tool
+        ) (lib.toList opt);
 
       setupOptions =
-        (mapAttrs (
+        (lib.mapAttrs (
           _:
           {
             linter ? [ ],
             formatter ? [ ],
           }:
-          (mkToolOption "linters" linter) ++ (mkToolOption "formatters" formatter)
-        ) (attrsets.filterAttrs (v: _: v != "all") cfg.setup))
+          (mkToolValue "linters" linter) ++ (mkToolValue "formatters" formatter)
+        ) (lib.attrsets.filterAttrs (v: _: v != "all") cfg.setup))
         // {
           "=" =
-            (mkToolOption "linters" cfg.setup.all.linter)
-            ++ (mkToolOption "formatters" cfg.setup.all.formatter);
+            (mkToolValue "linters" cfg.setup.all.linter) ++ (mkToolValue "formatters" cfg.setup.all.formatter);
         };
     in
-    mkIf cfg.enable {
+    lib.mkIf cfg.enable {
       extraPlugins = [ cfg.package ];
 
-      warnings = optional ((length nixvimPkgs.wrong) > 0) ''
+      warnings = lib.optional ((lib.length nixvimPkgs.wrong) > 0) ''
         Nixvim (plugins.efmls-configs): Following tools are not handled by nixvim, please add them to externallyManagedPackages to silence this:
-          ${concatStringsSep " " nixvimPkgs.wrong}
+          ${lib.concatStringsSep " " nixvimPkgs.wrong}
```

Comment:
Could you add a TODO comment about showing the offending `files` or `definitions`?

(Assuming you don't want to do that in this PR)

Reply:
I don't think I want to do it here, it does seem so easily to reverse the steps to find where those tools have been used, I'd have to refactor a lot of code I think. Adding a todo